hnn_core/network.py

```diff
@@ -251,8 +264,58 @@ def _add_external_feeds(self):
         unique_keys = sorted(self.p_unique.keys())
         self.feedname_list += unique_keys
 
-    def _create_gid_dict(self):
-        """Creates gid dicts."""
+        self._update_gid_dict()  # add feeds -> update the gid_dict
+        self.instantiate_feed_events()
+
+    def instantiate_feed_events(self):
```

Comment:
`instantiate_feeds` ?

I think names of python objects/variables are more precise than descriptive names. Because now I was wondering what is "feed events"

Reply:
OK. I'm also still wrapping my head around this. Was stoked that I managed to instantiate all `event_times`  prior to `NetworkBuilder`! Wasn't easy, but now `network_builder` no longer does _any_ random number generation at all.